lib/Optimizer/GraphOptimizer/OptimizationPipelines.cpp

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "glow/Optimizer/GraphOptimizer/GraphOptimizer.h"
+#include "glow/Optimizer/GraphOptimizer/PassManager.h"
+
+using namespace glow;
+
+FunctionPassPipeline glow::createDefaultGraphOptimizationPipeline() {
```

Comment:
So, you decided to stay with the design where pipelines are very visible and exist outside of PassManagers? As opposed to hiding them inside PMs and using something like 
```c++
glow::addDefaultGraphOptimizationPasses(FunctionPassManager &PM) {
   PM.addPass(FunctionPassConfig(...DCE...));
   PM.addPass(FunctionPassConfig(...SinleCode...));
   ...
}
```

Reply:
I pushed a commit doing this. I was thinking we would want more direct access to the pipelines for a backend to manipulate, but that's not the case for now -- a backend can only opt out of passes from the default fold/opt. There's still a getter for a const ref to the pipeline from the PM, not sure it's useful or necessary for now.